core/src/main/java/org/infinispan/AdvancedCache.java

```diff
@@ -169,6 +169,16 @@
    AuthorizationManager getAuthorizationManager();
 
    /**
+    * Whenever this cache acquires a lock it will do so using the given Object as the owner of said lock.
+    * <p>
+    * This method provides no guarantees as to how this affect locking and should not be used for general usage.
```

Comment:
That is part of the problem is we don't really have a nice way to have this on a non public API. But that is a different issue.

But I wouldn't call it a hack though, just there shouldn't ever be a reason that an end user should ever use this. I can detail what it does in more detail though, which should help curb someone from using it, instead of saying don't use it.